### DISCUSSION TITLE
How to remove dev integration for directory?

### DISCUSSION THREAD

**Discussion by acgetchell** [pkgxdev/pkgx#848] [Help]
`pkgx` dev integration borks the rust-analyzer toolchain:

```
Failed to find sysroot for Cargo.toml file /Users/adam/projects/cdt-rs/Cargo.toml. Is rust-src installed? can't load standard library from sysroot

/Users/adam/.pkgx/rust-lang.org/v1.73.0

(discovered via `rustc --print sysroot`)

try installing the Rust source the same way you installed rustc
```

I use `rustup` to manage my toolchains.

How do I remove the dev integration?

(Deleting and re-cloning the repo doesn't work.)

**Comment by xzone911** [comment]
you'll need to revert the changes made by `pkgx` and ensure your Rust environment is correctly set up using `rustup`. Here's a step-by-step guide:

1. Unset Environment Variables:
   - `pkgx` might have set some environment variables that are interfering with your Rust setup. Check your shell configuration files (like `.bashrc`, `.bash_profile`, `.zshrc`, etc.) and remove any lines that are related to `pkgx`. 

2. Revert Path Changes:
   - Ensure that the PATH variable has not been permanently modified to prioritize `pkgx` directories. Again, check your shell configuration files for any such modifications.

3. Reinstall Rust Toolchain:
   - Use `rustup` to reinstall your Rust toolchain. This can help reset any paths or settings that might have been altered.
   ```
   rustup toolchain uninstall <your-toolchain-version>
   rustup toolchain install <your-toolchain-version>
   ```

4. Install Rust Source:
   - Ensure that you have the Rust source files installed, as they are needed by rust-analyzer.
   ```
   rustup component add rust-src
   ```

5. Reconfigure rust-analyzer:
   - If rust-analyzer is still not working, try reconfiguring it or reinstalling it, depending on your editor setup.

6. Check for `.pkgx` Configurations:
   - Look for any `.pkgx` related files or configurations in your project directory and consider removing or modifying them if they're causing conflicts.

7. Clean and Rebuild:
   - After making these changes, it can be helpful to clean your project and rebuild it.
   ```
   cargo clean
   cargo build
   ```

8. Testing Rust Installation:
   - Test your Rust installation independently of your project to ensure it's working correctly. Try compiling a simple "Hello, World!" program in a new directory.

9. Consult pkgx Documentation or Support:
   - If these steps don't resolve the issue, refer to `pkgx` documentation or their support channels for specific instructions on undoing the dev integration.

Remember that the specifics can vary depending on your system configuration and how `pkgx` integrates with your environment. If these steps don't work, you might need to look for `pkgx`-specific uninstallation instructions or reach out to their support community.

**Comment by mxcl** [comment]
`dev off` undoes `dev`. This is documented, but I notice it is a little hidden so I'll improve it.

If you report the rust issue to pantry we'll fix that too. I can't quite tell from the output above what is going wrong.

**Reply by mxcl** [reply]
Sorry for the delay in response here. I hope you found a solution. `pkgx deactivate` would have worked as well. Or just deleting the pkgx binary.
